### DISCUSSION TITLE
Data Visualization

### DISCUSSION THREAD

**Discussion by noahmclean** [CIRDLES/Tripoli#37] [Ideas]
This thread is for ideas about mass spectrometer data (and metadata) visualization.  How do you want to see and interact with mass spectrometer data?

**Comment by noahmclean** [comment]
Sean Gaynor: isotope ratio vs. isotope ratio, even for isobaric interference and fractionation metrics.

**Comment by noahmclean** [comment]
Ryan Ickert: scaling can be a problem.  Original Tripoli sticks max and min at top and bottom.  Need a way to see bigger picture.

**Reply by noahmclean** [reply]
Noah: one idea is to add a text output for how fast a beam is growing or decaying.  Nicole: SHRIMP software has this and color-codes by beam decay rate.

**Reply by grahamedwards** [reply]
Sometimes I like to zoom in too! Maybe graphical way to scale axes so you can zoom in and out on the fly?

**Reply by noahmclean** [reply]
Great idea about the zooming.

**Comment by ryanickert** [comment]
Provide a graphical sense of scale on time series data - for example error bars based on counting stats (+/- on peak Johnson noise), or a static error bar on the side that indicates a sense of scale (e.g., 0.01%, 0.1%, 1% stacked error bars).

**Reply by noahmclean** [reply]
Clarification: this is for isotope ratio data time series.

**Reply by ryanickert** [reply]
I'll take it for signal intensities too...

**Reply by ryanickert** [reply]
Ok on second thought, DEFINITELY on signal intensities.  If I have a stable beam sometimes it's hard to tell if I'm just looking at shot noise or the beam has gone haywire, and putting error bars on them corresponding to counting stats +/- JN would let me know that things are good/bad.

**Reply by ryanickert** [reply]
![image](https://user-images.githubusercontent.com/7709328/176947164-93be6054-5837-4b6c-8481-d2b83e6881e5.png)

**Reply by ryanickert** [reply]
This is kind of what I mean.  This is peak-hopping Daly data from a mixed U-Pu run and I added the error bars myself (based on shot noise, ignoring interpolation complexities).

**Comment by noahmclean** [comment]
Graham: visualization that includes real-time outlier identification.  Recalculate after new data come in.

**Reply by noahmclean** [reply]
The crowd appreciates the ability to synchronize data rejection across all measurements.

**Reply by grahamedwards** [reply]
For consistency, relates to: https://github.com/CIRDLES/Tripoli/discussions/34#discussioncomment-2958283

**Comment by noahmclean** [comment]
Scott: provide some statistical help/test in figuring out when an isobaric interference exists or is gone.

**Comment by SeanPG36** [comment]
One quick idea would be to have the notes input in Tripoli read by Redux, so that information you noted during the ratio reduction could be easily visible from Redux.

**Comment by noahmclean** [comment]
Add annotations... as a time series (capture the analysis time) or assigned to a particular output stream -- intensity, isotope ratio, etc.

**Comment by noahmclean** [comment]
Have a pre-fabricated visualization organization, which can be saved and opened, and tied to a particular method.

**Comment by noahmclean** [comment]
Ryan and Sean: Maintain OG Tripoli functionality for visualization as well.  Drag and drop gets you plots and results.

**Comment by grahamedwards** [comment]
Histogram plot type. Ability to visualize certain blocks/groups of data and look at distributions at certain time windows or entire run.

**Reply by noahmclean** [reply]
Drag and select a subset of data and see a histogram for that data.

**Comment by noahmclean** [comment]
@ryanickert : plot everything in a simplex

**Comment by noahmclean** [comment]
@bowring : need virtual reality goggle interface, GUI

**Comment by noahmclean** [comment]
Add a text box for numerical outputs, that sits beside the other boxes for e.g. time series and any2 plots.

**Comment by noahmclean** [comment]
Scott: plot up the residuals to look for time dependence, etc.

**Comment by noahmclean** [comment]
Warren: having run tables handy is really helpful, and seeing statistics on the run tables.  Inspired by Ludwig's HobbsCalc.

**Comment by ryanickert** [comment]
Have a way to easily pop up the peak shape images while reducing data.

**Comment by grahamedwards** [comment]
Tracking/plotting peak centers, shoulders & baselines for each block.

**Comment by ryanickert** [comment]
Some way to vizualize the baselines/backgrounds.

**Comment by noahmclean** [comment]
Visualizations with an x-axis of mass: peak tails.  Ryan: peak tails.  Visualization of isobaric interference abundances.  Your own run.

**Comment by ryanickert** [comment]
Visualize ion beam intensities and interferences in mass vs. intensity

**Comment by noahmclean** [comment]
Warren: have. a big scale bar on intensity time series to show absolute value of beam growth/decay.

**Comment by ryanickert** [comment]
Barrel-level visualizations of on beam-decay, total ions, ionization efficiency.

**Comment by noahmclean** [comment]
Ryan: "barrel-level" stats like ionization efficiency, signal levels, etc.

**Reply by ryanickert** [reply]
https://github.com/CIRDLES/Tripoli/discussions/37#discussioncomment-3012877
